pkg/cmd/roachtest/cluster.go

```diff
@@ -1393,6 +1394,61 @@ func (c *clusterImpl) FetchDebugZip(
 	})
 }
 
+// FetchVMSpecs downloads the VM specs from the cluster using `roachprod get`.
+// The logs will be placed in the test's artifacts dir.
+func (c *clusterImpl) FetchVMSpecs(ctx context.Context, l *logger.Logger) error {
+	if c.IsLocal() {
+		return nil
+	}
+
+	l.Printf("fetching VM specs\n")
+	c.status("fetching VM specs")
+
+	vmSpecsFolder := filepath.Join(c.t.ArtifactsDir(), "vm_specs")
+	if err := os.MkdirAll(vmSpecsFolder, 0755); err != nil {
+		return err
+	}
+
+	// Don't hang forever if we can't fetch the specs.
+	return timeutil.RunWithTimeout(ctx, "fetch logs", 5*time.Minute, func(ctx context.Context) error {
+		cachedCluster, err := getCachedCluster(c.name)
+		if err != nil {
+			return err
+		}
+		providerToVMs := bucketVMsByProvider(cachedCluster)
+
+		var vmSpecs []map[string]interface{}
+		for provider, vms := range providerToVMs {
+			p := vm.Providers[provider]
+			vmSpecs, err = p.GetVMSpecs(l, vms)
+			if err != nil {
+				l.Errorf("failed to get VM spec for provider %s: %s", provider, err)
+				continue
+			}
+		}
+
+		for _, vmSpec := range vmSpecs {
```

Comment:
I'm only getting one node/file in `vm_specs` even when I run a multinode test. I did some brief debugging and `p.GetVmSpecs` does return all the vms in the cluster, but when we enter this loop, `vmSpecs` is always `len = 1`.